component/encryptor/encryptor.go

```diff
@@ -0,0 +1,382 @@
+
+	fileHandle, err := os.OpenFile(e.mountPointCipher+options.Name, os.O_RDWR|os.O_CREATE, options.Mode)
+	if err != nil {
+		log.Trace("Encryptor::createFile : Error creating file: %s", err.Error())
+		return nil, err
+	}
+
+	e.handle = fileHandle
+	// Set the file handle in the handle
+	return handle, nil
+}
+
+func formatListDirName(path string) string {
+	// If we check the root directory, make sure we pass "" instead of "/"
+	// If we aren't checking the root directory, then we want to extend the directory name so List returns all children and does not include the path itself.
+	if path == "/" {
+		path = ""
+	} else if path != "" {
+		path = internal.ExtendDirName(path)
+	}
+	return path
+}
+
+func (e *Encryptor) StreamDir(options internal.StreamDirOptions) ([]*internal.ObjAttr, string, error) {
+	var objAttrs []*internal.ObjAttr
+	path := formatListDirName(options.Name)
+	log.Info("Encryptor::StreamDir : %s", path)
+	// Get a list of files in the directory
+	files, err := os.ReadDir(e.mountPointCipher + path)
+	if err != nil {
+		log.Trace("Encryptor::StreamDir : Error reading directory %s : %s", path, err.Error())
+		return nil, "", err
+	}
+
+	// Iterate through files
+	for _, file := range files {
+		// Call GetAttr method for each file
+		attr, err := e.GetAttr(internal.GetAttrOptions{Name: path + file.Name()})
+		if err != nil {
+			if err != syscall.ENOENT {
+				log.Trace("Encryptor::StreamDir : Error getting file attributes: %s", err.Error())
+				return objAttrs, "", err
+			}
+			log.Trace("Encryptor::StreamDir : File not found: %s", file.Name())
+			continue
+		}
+
+		// Append the result to objAttrs
+		objAttrs = append(objAttrs, attr)
+	}
+
+	// Return the objAttrs list and nil error
+	return objAttrs, "", nil
+}
+
+func checkForActualFileSize(fileHandle *os.File, currentFileSize int64, blockSize int64) (int64, error) {
+
+	totalBlocks := currentFileSize / (blockSize + MetaSize)
+	if currentFileSize < totalBlocks*(blockSize+MetaSize)+8 {
+		return 0, nil
+	}
+
+	// Read the last 8 bytes of file and check for padding
+	paddingLengthBytes := make([]byte, 8)
+	_, err := fileHandle.ReadAt(paddingLengthBytes, currentFileSize-8)
+	if err != nil {
+		log.Err("Encryptor: Error reading last 8 bytes of file: %s", err.Error())
+		return 0, err
+	}
+
+	actualFileSize := currentFileSize - int64(binary.BigEndian.Uint64(paddingLengthBytes)) - MetaSize - 8
+	return actualFileSize, nil
+}
+
+func (e *Encryptor) GetAttr(options internal.GetAttrOptions) (attr *internal.ObjAttr, err error) {
+	// Open the file from the mount point and get the attributes.
+	log.Info("Encryptor::GetAttr for %s", options.Name)
+	fileAttr, err := os.Stat(e.mountPointCipher + options.Name)
+	if err != nil {
+		if os.IsNotExist(err) {
+			log.Trace("Encryptor::GetAttr : File not found: %s", options.Name)
+			return nil, syscall.ENOENT
+		}
+		log.Trace("Encryptor::GetAttr : Error getting file attributes: %s", err.Error())
+		return &internal.ObjAttr{}, nil
```

Comment:
If there is an error in this path then why don't we fail the call? Good to capture that as a comment.

Reply:
yup my bad. this should be returning error only.